src/snowplow/userEventHandler.ts

```diff
@@ -0,0 +1,144 @@
+import { buildSelfDescribingEvent } from '@snowplow/node-tracker';
+import { SelfDescribingJson } from '@snowplow/tracker-core';
+import { EventType, SnowplowEventMap, UserEventPayloadSnowplow } from './types';
+import { Account, ApiUser, ObjectUpdate, User } from './types';
+import { config } from '../config';
+import { EventHandler } from './EventHandler';
+import { tracker } from './tracker';
+
+type ObjectUpdateEvent = Omit<SelfDescribingJson, 'data'> & {
+  data: ObjectUpdate;
+};
+
+type AccountContext = Omit<SelfDescribingJson, 'data'> & {
+  data: Account;
+};
+
+type UserContext = Omit<SelfDescribingJson, 'data'> & {
+  data: User;
+};
+
+type ApiUserContext = Omit<SelfDescribingJson, 'data'> & {
+  data: ApiUser;
+};
+
+/**
+ * class to send `user-event` to snowplow
+ */
+export class UserEventHandler extends EventHandler {
+  constructor() {
+    super(tracker);
+    return this;
+  }
+
+  /**
+   * method to create and process event data
+   * @param data
+   */
+  async process(data: UserEventPayloadSnowplow): Promise<void> {
+    this.addRequestInfoToTracker(data);
+    const event = buildSelfDescribingEvent({
+      event: UserEventHandler.generateAccountUpdateEvent(data),
+    });
+    const context = UserEventHandler.generateEventContext(data);
+    await super.track(event, context);
```

Comment:
this isn't actually an async call.

Reply:
yes understood - we expect the ecs to run it at somepoint 

will add a comment to make this explicit